### PR TITLE
Feature/request caching mw

### PR DESCRIPTION
**Pull request by claylaut**

### CODE REVIEW

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			_serializer = serializer;
+			_cache = cache;
+		}
+
+		public async Task<FluentHttpResponse> Get(string hash)
+		{
+			var id = await hash.ComputeHash();
+			var result = await _cache.GetOrCreate(id, async _ =>
+			{
+				var item = await _clientDb.HttpResponses.FirstOrDefaultAsync(x => x.Id == id);
```
Comment (stephenlautier) [inline src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: use `FindAsync`

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	public class RemoteResponseCacheService : IResponseCacheService
+	{
+		private readonly FluentHttpClientDbContext _clientDb;
+		private readonly IHttpResponseSerializer _serializer;
+		private readonly IMemoryCache _cache;
+
+		public RemoteResponseCacheService(
+			FluentHttpClientDbContext clientDb,
+			IHttpResponseSerializer serializer,
+			IMemoryCache cache)
```
Comment (stephenlautier) [inline src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: brace formatting

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+using FluentlyHttpClient.Caching;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+using Sketch7.Core;
+
+namespace FluentlyHttpClient.Entity
+{
+	public class RemoteResponseCacheService : IResponseCacheService
+	{
+		private readonly FluentHttpClientDbContext _clientDb;
```
Comment (stephenlautier) [inline src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: rename `_clientDb` => `dbContext`

File: src/FluentlyHttpClient/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -20,8 +21,12 @@ public static IServiceCollection AddFluentlyHttpClient(this IServiceCollection s
 		{
 			if (services == null) throw new ArgumentNullException(nameof(services));
 
+			services.TryAddSingleton<IHttpResponseSerializer, HttpResponseSerializer>();
 			services.TryAddSingleton<IFluentHttpClientFactory, FluentHttpClientFactory>();
 			services.TryAddTransient<FluentHttpMiddlewareBuilder>();
+			services.TryAddSingleton<IResponseCacheService, MemoryResponseCacheService>();
+			services.AddMemoryCache();
+			services.TryAddTransient<FluentHttpMiddlewareBuilder>();
```
Comment (stephenlautier) [inline src/FluentlyHttpClient/ServiceCollectionExtensions.cs]: remove - duplicate

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs (hunk 1 above)
Comment (claylaut) [reply src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: done

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs (hunk 3 above)
Comment (claylaut) [reply src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: done

File: src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs (hunk 2 above)
Comment (claylaut) [reply src/FluentlyHttpClient.Entity/RemoteResponseCacheService.cs]: done

File: src/FluentlyHttpClient/ServiceCollectionExtensions.cs (hunk 4 above)
Comment (claylaut) [reply src/FluentlyHttpClient/ServiceCollectionExtensions.cs]: done

**Pull request merged**